test/facade_converter_test.ts

```diff
@@ -237,6 +243,99 @@ var y = x.length;`);
 }`);
   });
 
+  describe('promises', () => {
+    it('translates into Futures', () => {
+      expectWithTypes('let x: Promise = Promise.resolve(1);').to.equal(`import "dart:async";
+
+Future x = Future.value(1);`);
+      expectWithTypes('let x: Promise = Promise.reject(1);').to.equal(`import "dart:async";
+
+Future x = Future.error(1);`);
+      expectWithTypes('let x: Promise = new Promise((resolve) => {resolve(1);});')
+          .to.equal(`import "dart:async";
+
+Future x = (() {
+  Completer _completer$$ts2dart$0 = new Completer();
+  var resolve = _completer$$ts2dart$0.complete;
+  (() {
+    resolve(1);
+  })();
+  return _completer$$ts2dart$0.future;
+})();`);
+      expectWithTypes('let x: Promise = new Promise((resolve, reject) => {resolve(1);});')
+          .to.equal(`import "dart:async";
+
+Future x = (() {
+  Completer _completer$$ts2dart$0 = new Completer();
+  var resolve = _completer$$ts2dart$0.complete;
+  var reject = _completer$$ts2dart$0.completeError;
+  (() {
+    resolve(1);
+  })();
+  return _completer$$ts2dart$0.future;
+})();`);
+      expectWithTypes('let x: Promise = new Promise((myParam1, myParam2) => {myParam1(1);});')
+          .to.equal(`import "dart:async";
+
+Future x = (() {
+  Completer _completer$$ts2dart$0 = new Completer();
+  var myParam1 = _completer$$ts2dart$0.complete;
```

Comment:
Kind of sad for angular dart users that the library sources they browse are now more obviously generated code